### ISSUE TITLE
Common docstring formats are not displayed nicely in docstring hover pop-up

### ISSUE THREAD

**Issue by shaypal5** [DonJayamanne/pythonVSCode#821]
## Environment data
VS Code version:  1.10.2
Python Extension version:  6.0
Python Version:   3.5.3
OS and version:    OS X El Capitan Version 10.11.6

## Actual behavior
When you hover over a function call in your code, you get a small pop-up showing you the docstring of that function. Function signature is formatted nicely, and reStructured Text headers are respected, but elements such as parameter and/or return value names and types - as formatted according to common conventions such as [Numpy's](https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt), [Google's](http://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html) or [Sphnix](http://thomas-cokelaer.info/tutorials/sphinx/docstring_python.html) - are shown in raw text form.

## Expected  behavior
Docstrings using the above format should be formatted nicely.

## Steps to reproduce:  
-  Write a docstring for a function in one of the following formats: Numpy, Google, Sphinx.
-  Hover over a call to that function.

## Comments
I don't know if this functionality comes from general VSCode code or this extension's code. In any case, I'd really appreciate getting a pointer to the right direction so I can attempt to contribute an implementation of this feature.

**Comment by gpassero** [comment]
Example of docstring in VS:

![image](https://cloud.githubusercontent.com/assets/4478523/24613895/0d8df93c-1860-11e7-94b9-f885578f1369.png)

Same in VS Code:

![image](https://cloud.githubusercontent.com/assets/4478523/24613923/1f5909e0-1860-11e7-9144-1bc0f66939da.png)

**Comment by DonJayamanne** [comment]
@shaypal5 Please could you confirm this is still an issue in the latest version of VS Code.

**Comment by gandhis1** [comment]
Is it possible to remove the redundant function declaration without syntax highlighting and also reduce the font size on the docstring contents itself? This would make the hover box more compact and readable, in my view.

**Comment by hirschmann** [comment]
VS Code version: 1.11.0
Python Extension version: 0.6.0

I can confirm this is still an issue. No improvements compared to VS Code version 1.10.2.

**Comment by shaypal5** [comment]
Same here.

@DonJayamanne 
I took a look at the code a few days ago and found the function in charge, and did a crash course in TypeScript. I'll try to get to this in a few days. 

I'll start with the redundant function declaration that @gandhis1 mentioned, as I've already found the source of the problem in the code, and do a separate pull request for it. It'll give me a chance to play with the code and testing with a simpler problem, and will separate the bug fix from the feature.

**Comment by neilsustc** [comment]
@shaypal5 Could you please have a look at #888 at the same time?

If the docstring ends with `)`, it won't be displayed at all.

**Comment by shaypal5** [comment]
@neilsustc 

Sure. Hope to get to it soon.

**Comment by neilsustc** [comment]
Sorry for not updating you 😂. I have made a PR to fix #888 right now, just a little change.

---

Coming back to this issue, it seems that vscode renders hover text with Markdown syntax. But there is a gap between Markdown and reStructuredText which is used in Python doc.

Take this as an simplified example,

```python
def loadtxt(fname, dtype=float, comments='#', delimiter=None):
    """
    Load data from a text file.

    Parameters
    ----------
    fname : file or str
        File, filename, or generator to read.  If the filename extension is
        ``.gz`` or ``.bz2``, ...

    Returns
    -------
    out : ndarray
        Data read from the text file.

    Notes
    -----
    This function aims to be a fast reader for simply formatted files. ...

    .. versionadded:: 1.10.0

    Examples
    --------
    >>> from io import StringIO   # StringIO behaves like a file object
    >>> c = StringIO("0 1\\n2 3")
    >>> np.loadtxt(c)
    array([[ 0.,  1.],
           [ 2.,  3.]])

    """
    pass
```

Here are some blocks to think about,

1. `fname : file or str`. I think we can just leave it literally because it seems not a RST syntax.
2. The indention means 'definition list' in RST. How about change it to Markdown syntax?
3. \`\`.gz\`\`. It means inline literals. But there is no need to do any transformation.
4. `.. versionadded:: 1.10.0.`. I don't have a idea. [RST directives](http://docutils.sourceforge.net/docs/ref/rst/restructuredtext.html#directives)
5. `>>> from io import ...`. Doctest blocks. I prefer to put all in one Markdown code block (the current implementation put every `>>>` line in separate code block and remove the `>>>`).

How do you think of it?
